src/main/java/ch/njol/skript/events/EvtTabComplete.java

```diff
@@ -0,0 +1,72 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser;
+import org.bukkit.event.Event;
+import org.bukkit.event.server.TabCompleteEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+public class EvtTabComplete extends SkriptEvent {
```

Comment:
I think tab completion is out of scope for this PR - it should ideally be separated into separate unknown command and tab completion PRs.
I also think more care should be taken with supporting tab completion, especially when working with Skript commands.

Reply:
tab complete was the original intent for this, however after a few changes it ended up becoming a bit separated I can agree on that when I push the other stuff I'll see about removing the tab complete and move it into it's own pr when I wake up later